lang/py/src/avro/io.py

```diff
@@ -478,6 +573,77 @@ def write_crc32(self, bytes):
     """
     self.write(STRUCT_CRC32.pack(crc32(bytes) & 0xffffffff));
 
+  def write_date_int(self, datum):
+    """
+    Encode python date object as int.
+    It stores the number of days from
+    the unix epoch, 1 January 1970 (ISO calendar).
+    """
+    delta_date = datum - datetime.date(1970, 1, 1)
+    if delta_date.days < 0:
```

Comment:
The negative date value can be valid, which represents the timeline before 1/1/1970.